Merge 15.1 code freeze into trunk

spencertransier opened the pull request:
This PR merges the changes from the 15.1 code freeze into `trunk`. Includes: 
 - Version bump (`15.1.0.1` because there was an issue with the initial build and I had to create a new build: p1693007584312879/1693004877.995529-slack-C5ALGJYEP )
 - Updated release notes
 - Frozen app strings for translation
 - Added missing xcconfig env var: https://github.com/woocommerce/woocommerce-ios/pull/10544

spencertransier committed:
Update draft release notes for 15.1.

spencertransier committed:
Bump version number

spencertransier committed:
Release Notes: add new section for next version (15.2)

spencertransier committed:
Update release notes for 15.1

spencertransier committed:
Update metadata strings

spencertransier committed:
Freeze strings for localization

spencertransier committed:
Add missing PUBLIC_CONFIG_FILE env var

spencertransier committed:
Merge pull request #10544 from woocommerce/add/missing-xcconfig-env-var

Add missing `PUBLIC_CONFIG_FILE` env var

spencertransier committed:
Bump version number

mokagio commented on fastlane/Fastfile:
```diff
@@ -15,7 +15,8 @@ GITHUB_REPO = 'woocommerce/woocommerce-ios'
 PROJECT_ROOT_FOLDER = File.dirname(File.expand_path(__dir__))
 RESOURCES_FOLDER = File.join(PROJECT_ROOT_FOLDER, 'WooCommerce', 'Resources')
-PUBLIC_VERSION_CONFIG_FILE = File.join(PROJECT_ROOT_FOLDER, 'config', 'Version.Public.xcconfig')
+PUBLIC_CONFIG_FILE = File.join(PROJECT_ROOT_FOLDER, 'config', 'Version.Public.xcconfig')
+ENV['PUBLIC_CONFIG_FILE'] = PUBLIC_CONFIG_FILE
```
@spencertransier  Did you decide to rename `PUBLIC_VERSION_CONFIG_FILE` to `PUBLIC_CONFIG_FILE` so that it has the same name as the environment variable?

spencertransier replied:
@mokagio Yep, while I was re-adding the environment variable, I decided to rename the constant to match what you did in https://github.com/wordpress-mobile/WordPress-iOS/commit/86b03cb994b6f8540ea903fde712f5dab179c1ab for a bit of added consistency across apps.

mokagio committed:
Temporarily switch release-toolkit build to address `.pot` generation

There's been a removal of `\n` characters that, while probably
harmless, might result in unnecessary translation noise.

mokagio committed:
Update metadata strings

mokagio committed:
Revert "Temporarily switch release-toolkit build to address `.pot` generation"

This reverts commit a58acd844da6afac35e0b5ea8f02d83acee8815c.

We ought to fix the problem at the root, not risk using an out of date
version of the release-toolkit.

mokagio commented on WooCommerce/Resources/AppStoreStrings.pot:
```diff
@@ -87,31 +87,31 @@ msgstr ""
 msgid ""
 "Get notified of\n"
-"every sale\n"
+"every sale"
```
The various `\n` removals were unexpected. The linter didn't complain so I think they'd be effectively harmless. Still, here's a PR to fix them before merging on `trunk` to avoid noise in GlotPress.

https://github.com/woocommerce/woocommerce-ios/pull/10545

spencertransier replied:
Thank you for putting up that PR! I merged it into `release/15.1` then updated the merge branch that this PR is based on.

mokagio reviewed (approved):
Approving to unblock but I think we should merge https://github.com/woocommerce/woocommerce-ios/pull/10545 into here first @spencertransier

spencertransier committed:
Merge pull request #10545 from woocommerce/mokagio/fix-metadata-po

Restore `\n` removed from `.pot` file
